sway/commands.c

```diff
@@ -1401,17 +1401,33 @@ struct cmd_results *handle_command(char *_exec) {
 
 	head = exec;
 	do {
-		// Handle criteria
+		// Extract criteria (valid for this command list only).
+		criteria = NULL;
 		if (*head == '[') {
+			++head;
 			criteria = argsep(&head, "]");
 			if (head) {
 				++head;
 				// TODO handle criteria
 			} else {
-				results = cmd_results_new(CMD_INVALID, NULL, "Unmatched [");
+				if (!results) {
+					results = cmd_results_new(CMD_INVALID, criteria, "Unmatched [");
+				}
+				goto cleanup;
 			}
 			// Skip leading whitespace
 			head += strspn(head, whitespace);
+
+			// TODO: it will yield unexpected results to execute commands
+			// (on any view) that where meant for certain views only.
+			if (!results) {
+				char *tmp = malloc(128);
```

Comment:
Do not allocate a fixed buffer - measure the size of the final string and allocate the correct amount of memory.